src/io/buffered.cr

```diff
@@ -179,6 +179,33 @@ module IO::Buffered
     end
   end
 
+  # Returns the current position (in bytes) in this `IO`.
+  #
+  # ```
+  # File.write("testfile", "hello")
+  #
+  # file = File.new("testfile")
+  # file.pos     # => 0
+  # file.gets(2) # => "he"
+  # file.pos     # => 2
+  # ```
+  def pos : Int64
+    in_rem = @in_buffer_rem.size
+    out_rem = @out_count
+    rem = if in_rem > 0
+            raise RuntimeError.new("can't mix read + write when .sync = false") if out_rem > 0
+            in_rem
+          else
+            out_rem * -1
+          end
+
+    if responds_to?(:unbuffered_pos)
+      unbuffered_pos - rem
```

Comment:
```suggestion
    if self.responds_to?(:unbuffered_pos)
      self.unbuffered_pos - rem
```

Reply:
Hi, I don't understand this one.  When should `self` be used for method calls?  Is there an idiomatic crystal guide that shows when?

Reply:
If you use `self`, then after `responds_to?` `self` will be restricted to the types that respond to that method. If you don't use `self`, that doesn't happen. That means that if you try this code with a type that doesn't respond to `unbuffered_pos`, it will simply not compile. So the suggestion by @Sija is pretty much correct.

That probably means a spec is lacking for types where `unbuffered_pos` isn't defined.

Reply:
Or maybe that's already covered because CI is failing.

Reply:
Yes, that's the reason for the CI failure. So it's properly detected.

Reply:
Note that I described the current behavior. I don't know if that's ideal (maybe all calls should be restricted after that regardless of whether there's a `self.` or not), but that's the way things are right now.